controllers/storagecluster/storagecluster_controller_test.go

```diff
@@ -1139,3 +1141,44 @@ func getSemVer(version string, majorDiff uint64, islower bool) string {
 
 	return sv.String()
 }
+
+type ListenServer struct {
+	Listener net.Listener
+}
+
+func NewListenServer(endpoint string) (*ListenServer, error) {
+	ls := &ListenServer{}
+
+	rxp := regexp.MustCompile(`^http[s]?://`)
+	// remove any http or https protocols from the endpoint string
+	endpoint = rxp.ReplaceAllString(endpoint, "")
+	ln, err := net.Listen("tcp4", endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	ls.Listener = ln
+
+	go func(ln net.Listener) {
+		for {
+			conn, err := ln.Accept()
+			if err != nil {
+				break
+			}
+			conn.Close()
+		}
+	}(ls.Listener)
+
+	return ls, nil
+
+}
+
+func startServerAt(t *testing.T, endpoint string) {
+	ls, err := NewListenServer(endpoint)
+	if err != nil {
+		t.Logf("Failed to start ListenServer: %v", err)
+		return
```

Comment:
Would it be better to panic the test (or `t.Fatal`) upon this failure since it will cause some tests to fail anyway?

Reply:
I thought about this one a fair amount. The main issue is that there is *one* test case that purposefully supplies and invalid port. It's not the best, but to change that I would've had to refactor even more auxiliary functions that I did not feel like diving into. :joy: Also, in general, if for some reason the ListenServer fails to start, something else in the test is probably going to break anyway. This was my compromise to at least always highlight it even if it's expected.